File: packages/marie-plugin-document-extraction/tests/fixtures/sample.java
package com.example.invoice;

import java.util.List;

public class InvoiceParser {

    public interface LineItem {
        String description();

        double amount();
    }

    private final String currency;

    public InvoiceParser(String currency) {
        this.currency = currency;
    }

    public double totalAmount(List<LineItem> items) {
        return items.stream().mapToDouble(LineItem::amount).sum();
    }
}

File: packages/marie-plugin-document-extraction/tests/fixtures/sample.ts
import { readFileSync } from "node:fs";

export interface LineItem {
  description: string;
  amount: number;
}

export enum InvoiceStatus {
  Draft,
  Sent,
  Paid,
}

export type InvoiceId = string;

export class InvoiceParser {
  constructor(private readonly currency: string = "USD") {}

  parse(path: string): LineItem[] {
    return JSON.parse(readFileSync(path, "utf-8"));
  }
}

export function totalAmount(items: LineItem[]): number {
  return items.reduce((sum, item) => sum + item.amount, 0);
}
